packages/react/tests/index.test.tsx

```diff
@@ -337,4 +337,58 @@ describe('styled', () => {
     });
     expect(renderer.create(<Button size={0}>height should equal '1px'</Button>).toJSON()).toMatchSnapshot();
   });
+
+  test('it matches compound variants with multiple breakpoints', () => {
+    const { styled: _styled, css: _css } = createStyled({
+      showFriendlyClassnames: true,
+      breakpoints: {
+        breakpointOne: (rule) => `@media(min-width:400px){${rule}}`,
+        breakpointTwo: (rule) => `@media(min-width:800px){${rule}}`,
+      },
+    });
+
+    const Button = _styled('button', {
+      color: 'gray',
+
+      variants: {
+        color: {
+          red: {
+            background: 'red',
+          },
+          yellow: {
+            background: 'yellow',
+          },
+        },
+        size: {
+          small: {
+            height: 20,
+          },
+          big: {
+            height: 40,
+          },
+        },
+      },
+    });
+
+    Button.compoundVariant(
+      {
+        color: 'red',
+        size: 'big',
+      },
+      {
+        backgroundColor: 'potato',
+      }
+    );
+
+    expect(
+      renderer
+        .create(
+          <Button color={{ initial: 'red', breakpointOne: 'yellow' }} size={{ breakpointTwo: 'big' }}>
+            Responsive compound variants
+          </Button>
+        )
```

Comment:
Nice! this works as I would expect

This test is a bit complex though, since it relies on an implicit breakpoint match.

Do you think its worth writing a few different tests for the compound API, for the sake of clarity?

- the most basic usage
- a usage where the breakpoint match is explicit
- a usage where the breakpoint match is implicit (this one)

Reply:
Updated the tests 👍